dbms/src/Functions/FunctionsStringSearch.h

```diff
@@ -16,13 +16,92 @@
 
 #include <Columns/ColumnConst.h>
 #include <Columns/ColumnString.h>
+#include <Columns/IColumn.h>
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+#include <Common/UTF8Helpers.h>
+#include <Common/typeid_cast.h>
 #include <DataTypes/DataTypeString.h>
 #include <DataTypes/DataTypesNumber.h>
 #include <Functions/FunctionHelpers.h>
 #include <Functions/IFunction.h>
+#include <Functions/StringUtil.h>
+
+#include <cstring>
+#include <string_view>
+#include "common/defines.h"
 
 namespace DB
 {
+using Chars_t = ColumnString::Chars_t;
+using Offsets = ColumnString::Offsets;
+
+struct IlikeHelper
+{
+    static void lowerStrings(Chars_t & chars)
+    {
+        size_t size = chars.size();
+        size_t i = 0;
+        while (i < size)
+        {
+            if (unlikely(chars[i] == '\0'))
```

Comment:
`isASCII('\0')` returns true, so I think we don't need to handle `\0` here?

Reply:
okk